src/main/java/bookstore/validation/PasswordValidator.java

```diff
@@ -0,0 +1,14 @@
+package bookstore.validation;
+
+import bookstore.dto.user.UserRegistrationRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class PasswordValidator implements ConstraintValidator<FieldMatch, Object> {
+
+    @Override
+    public boolean isValid(Object obj, ConstraintValidatorContext constraintValidatorContext) {
+        UserRegistrationRequestDto user = (UserRegistrationRequestDto) obj;
+        return user.getPassword().equals(user.getRepeatPassword());
```

Comment:
Can getPassword() return `null`? 
If it will return null, how are we going to handle NPE?

Reply:
Why then we use @NotBlank if we need to check NPE still?

Reply:
No one guarantees that this annotation will not be deleted.
Just use [Objects#equals](https://www.baeldung.com/java-comparing-objects#equals-static).